### ISSUE TITLE
All metrics computing a  curve should always send back result in the order Xaxis - Yaxis - Everything_Else 

### ISSUE THREAD

**Issue by Yann-CV** [Lightning-AI/torchmetrics#2200]
## 🐛 Bug

The metrics giving access to curve are not always returning Xaxis - Yaxis in the same order (considering the common usage of the curve).

### To Reproduce

```
pred = torch.tensor([0, 0.1, 0.8, 0.4])
target = torch.tensor([0, 1, 1, 0])

roc = ROC(task="binary")
fpr, tpr, thresholds = roc(pred, target) # in ROC curve Xaxis = fpr -------- Yaxis = tpr

pr_curve = PrecisionRecallCurve(task="binary")
precision, recall, thresholds = pr_curve(pred, target) # in precision recall curve Xaxis = recall -------- Yaxis = precision
```

### Expected behavior

```
pred = torch.tensor([0, 0.1, 0.8, 0.4])
target = torch.tensor([0, 1, 1, 0])

roc = ROC(task="binary")
fpr, tpr, thresholds = roc(pred, target) # in ROC curve Xaxis = fpr -------- Yaxis = tpr

pr_curve = PrecisionRecallCurve(task="binary")
recall, precision, thresholds = pr_curve(pred, target) # in precision recall curve Xaxis = recall -------- Yaxis = precision
```

### Environment

TorchMetrics 1.2

### Additional context

We are trying to be metric agnostic during automated plot creation. 
So far we added a constant object:

```
METRIC_AXIS = {
     ROC: {"x": 0, "y": 1},
     PrecisionRecallCurve: {"x": 1, "y": 0},
}
```

**Comment by SkafteNicki** [comment]
The goal here has been to be consistent with the implementations in sklearn:
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html
which has this ordering.

**Comment by Yann-CV** [comment]
@SkafteNicki  see https://github.com/Lightning-AI/torchmetrics/issues/2207#issuecomment-1825671216 (thanks again for the answer)
